routes/chat.js

```diff
@@ -9,14 +9,20 @@ const SuccessMiddleware = require('../middlewares/success');
 const sendSignedMessage = require('../controllers/chat/sendSignedMessage');
 const setSignedChannelAsRead = require('../controllers/chat/setSignedChannelAsRead');
 const BodyValidationMiddleware = require('../middlewares/body-validation');
+const getSignedChannelList = require('../controllers/chat/getSignedChannelList');
 
 router.get('/create-channel', chatController.createChannel);
 router.post('/add-moderator', chatController.addChannelModerator);
 router.post('/add-members-channel', auth.isAuth, chatController.addMembers);
 router.post('/anonymous', auth.isAuthAnonim, chatController.sendAnonymous);
 router.post('/send-signed-message', auth.isAuthV2, sendSignedMessage);
 router.get('/channels', auth.isAuthAnonim, chatController.getChannels);
-router.get('/channels/signed', auth.isAuthV2, chatController.getChannels);
+router.get(
+  '/channels/signed',
+  BodyValidationMiddleware.commonLimitOffset,
+  auth.isAuthV2,
+  getSignedChannelList
+);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new route `/channels/signed` has been added with middleware for body validation and authentication. It's good to see that the `commonLimitOffset` middleware is being used for validating limit and offset parameters, and `auth.isAuthV2` middleware is being used for authentication. However, it would be better if you could add error handling middleware at the end of this route to handle any errors that might occur in the `getSignedChannelList` controller. This will ensure that your application does not crash due to unhandled errors.
```diff
  router.get(
    '/channels/signed',
    BodyValidationMiddleware.commonLimitOffset,
    auth.isAuthV2,
    getSignedChannelList,
+   errorHandlerMiddleware
  );
```
Please replace `errorHandlerMiddleware` with the actual error handling middleware used in your application.


<!-- This is an auto-generated comment by OSS CodeRabbit -->